.github/actions/create-release/action.yml

```diff
@@ -0,0 +1,140 @@
+# limitations under the License.
+
+#
+# This GitHub Action creates a Git tag and associated GitHub Release at the current SHA.
+#
+# It includes two outputs: the "current_version" and the "next_version".
+#
+# Example usage:
+#
+#     jobs:
+#       release:
+#         runs-on: 'ubuntu-latest'
+#
+#         # The following line is super super super important! This ensures the
+#         # job only runs when the commit matches the expected author and
+#         # message. There are additional checks in the action itself, including
+#         # commit signature verification, but adding this line prevents the
+#         # workflow from running at all, saving your previous GitHub Actions
+#         # usage minutes.
+#         if: '${{ startsWith(github.event.head_commit.message, ''Release: v'') && github.event.head_commit.author.email == ''user_or_bot@company.com'' }}'
+#
+#         steps:
+#           - name: 'Create release'
+#             id: 'create-release'
+#             uses: 'abcxyz/pkg/.github/actions/create-release@main'
+#             with:
+#               github_token: '${{ secrets.GITHUB_TOKEN }}' # Or an organization PAT
+#
+#           # Example using go-releaser
+#           - uses: 'actions/checkout@v4'
+#             with:
+#               fetch-depth: 0
+#
+#           - uses: 'actions/setup-go@v5'
+#             with:
+#               go-version: '1.21'
+#
+#           - uses: 'goreleaser/goreleaser-action@v5'
+#             with:
+#               args: 'release --clean'
+#             env:
+#               GORELEASER_CURRENT_TAG: 'v${{ steps.create-release.outputs.release_version }}'
+#               GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}' # Or an organization PAT
+#
+
+name: 'create-release'
+
+inputs:
+  github_token:
+    description: |-
+      GitHub PAT or App Token to use for authentication.
+    type: 'string'
+    required: true
+
+  draft:
+    description: |-
+      Create the release as a draft.
+    type: 'boolean'
+    required: false
+    default: false
+
+outputs:
+  release_version:
+    description: |-
+      Version that is being released, without a leading "v".
+    value: '${{ steps.create-release.outputs.release_version }}'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Create release'
+      id: 'create-release'
+      env:
+        DRAFT: '${{ inputs.draft }}'
+      uses: 'actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea' # ratchet:actions/github-script@v7
+      with:
+        github-token: '${{ inputs.github_token }}'
+        script: |-
+          // Ensure the commit is signed.
+          const commitResult = await github.rest.repos.getCommit({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: context.payload.head_commit.id,
+          })
+          if (!commitResult.data.commit.verification.verified) {
+            core.setFailed(`Commit is not signed`)
+            return;
+          }
```

Comment:
Do we need this? I feel like the ruleset should enforce this?

Reply:
Probably not? When this was a reusable workflow, this was ensuring that the email address wasn't forged.

Reply:
Actually, we do need this. It's possible to sign a commit as a bot (which will pass rulesets), but fail validation. This isn't just checking that the commit is signed, but it's also checking that the commit is _verified_; rulesets don't do that as far as I can tell.